index.js

```diff
@@ -18,18 +18,17 @@ function koaDevware (compiler, options) {
       compiler.plugin('failed', (error) => {
         reject(error);
       });
-
-      dev(context.req, {
-        end: (content) => {
-          context.body = content;
-        },
-        setHeader: context.set.bind(context)
-      }, next);
     });
   }
 
   return async (context, next) => {
-    await middleware(context, next);
+    await waitUntilValidOrFailed();
+    await dev(context.req, {
```

Comment:
as @wtgtybhertgeghgtwtg mentioned, `dev` doesn't return a promise and will resolve immediately because of that. we ran into that same problem in an earlier version.